scripts/vcpkg_triplet.sh

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env bash
+
+##  This file is part of scilla.
+##
+##  Copyright (c) 2022 - present Zilliqa Research Pvt. Ltd.
+##  
+##  scilla is free software: you can redistribute it and/or modify it under the
+##  terms of the GNU General Public License as published by the Free Software
+##  Foundation, either version 3 of the License, or (at your option) any later
+##  version.
+## 
+##  scilla is distributed in the hope that it will be useful, but WITHOUT ANY
+##  WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+##  A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+## 
+##  You should have received a copy of the GNU General Public License along with
+##  scilla.  If not, see <http://www.gnu.org/licenses/>.
+
+# Determine vcpkg os and arch triplet
+OS="unknown"
+ARCH="$(uname -m)"
+
+unameOut="$(uname -s)"
+case "${unameOut}" in
+    Linux*)     OS=linux;;
+    Darwin*)    OS=osx;;
+    *)          echo "Unknown machine ${unameOut}"
+esac
+
+unameOut="$(uname -m)"
+case "${unameOut}" in
+    arm*)       ARCH=arm64;;
+    x86_64*)    ARCH=x64;;
+    *)          echo "Unknown machine ${unameOut}"
```

Comment:
I suggest to exit early in this case returning an error code.

Reply:
Good point, but it won't mean much here actually because this is a small utility used by OCaml to determine the vcpkg triplet. Whichever wrong triplet it returns, the build will fail immediately because no such directory will exist.

Reply:
I see. But I would prefer to make error handling more explicit, since the logic of this script and the OCaml program can be changed in the future.